server/index.js

```diff
@@ -2,20 +2,87 @@ require('dotenv/config');
 const path = require('path');
 const express = require('express');
 const errorMiddleware = require('./error-middleware');
+const pg = require('pg');
+const argon2 = require('argon2');
+const jwt = require('jsonwebtoken');
+const ClientError = require('./client-error');
+const authorizationMiddleware = require('./authorization-middleware');
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
 
 const app = express();
 const publicPath = path.join(__dirname, 'public');
+app.use(express.json());
 
 if (process.env.NODE_ENV === 'development') {
   app.use(require('./dev-middleware')(publicPath));
 } else {
   app.use(express.static(publicPath));
 }
 
-app.get('/api/hello', (req, res) => {
-  res.json({ hello: 'world' });
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
```

Comment:
Since your frontend is requiring an 8 character minimum on the password you want to check that here also. Front end validation is just for ease of use for the user but is easy to circumnavigate. Validation is most important on the server, people can send requests to your server without your front end, so it is important that the server validates all data it receives.